src/types/message.ts

```diff
@@ -0,0 +1,36 @@
+
+import { hrtime } from 'node:process';
+
+import { Table } from '../db/db';
+
+export type CrdtMessage = {
+  timestamp: string;
+  entityId: string;
+  table: Table;
+  column: string;
+  value: string;
+}
+
+export type ProcessedMessage = {
+  messageId: string;
+  table: Table;
+  column: string;
+  entityId: string;
+  processedAt: string;
+}
+
+type Values<T> = {
+  id: string;
+} & Partial<T>
+
+export function getCrdtUpdateMessages<T>(table: Table, values: Values<T>): CrdtMessage[]{
+  const messages: CrdtMessage[] = Object.keys(values).filter(key => key !== 'id').map(key => ({
+    timestamp: hrtime.bigint().toString(),
```

Comment:
yeh lets just use normal timestamps, the same as we do in other parts of the codebase

Reply:
normal timestamps didn't have enough resolution to be unique and used as an ID in this case, so I opted for nanoseconda as an easy solution.  In the future I think this should have a logical time component and a physical time component so we should use something like an HLC, but this was a quick way to get it working for now so I could begin testing some things